src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/hmi/rc_get_interior_vehicle_data_consent_request.cc

```diff
@@ -45,7 +45,10 @@ RCGetInteriorVehicleDataConsentRequest::RCGetInteriorVehicleDataConsentRequest(
                                                   params.application_manager_,
                                                   params.rpc_service_,
                                                   params.hmi_capabilities_,
-                                                  params.policy_handler_) {}
+                                                  params.policy_handler_) {
```

Comment:
Shouldn't this be read from smartDeviceLink.ini config file. Why is this timeout doubled?

Reply:
@atiwari9
Yes, initially default timeout value is taken from the SDL config file for all command requests:
https://github.com/smartdevicelink/sdl_core/blob/60e3354b17bcecd20d7727c71a58b34ba81f4af0/src/components/application_manager/src/commands/command_impl.cc#L84

Later, some particular commands are adjusting the default timeout according to their internal needs.
In this case, timeout is doubled to give HMI extra time for showing user consent prompt popup (which is another +10 seconds) - please look at linked issue #2829 for more details and sequence diagram.

Reply:
I understand that this may have been used as a work around earlier, when HMI probably didn't have a way to reset the timeout on SDL side, now it does. So we really don't and shouldn't still add such patches in SDL. If this is done to maintain the legacy implementation, i understand. But even in that case please log a bug/improvement item so that the onus can now be shifted back to HMI. After all, that's the whole point of enabling HMI to reset timeout for RPCs.

Reply:
@atiwari9 ok, I think we can remove all these timeout adjustments from existing PR and rely on HMI to control the lifetime of corresponding requests properly. In that case, we need the following changes:
1. Remove timeout adjustment changes from affected SDL core requests
2. Update the same requests on the HMI side - if HMI notices that extra time from the user is required then HMI sends OnResetTimeout to SDL with a doubled timeout value so SDL can wait for enough time
3. Check with ATF scripts that updated implementation does not break the legacy code and feature works as expected

Please let me know, is that plan works for you?